src/hotspot/share/gc/serial/defNewGeneration.cpp

```diff
@@ -981,27 +981,18 @@ bool DefNewGeneration::no_allocs_since_save_marks() {
   return to()->saved_mark_at_top();
 }
 
-void DefNewGeneration::contribute_scratch(ScratchBlock*& list, Generation* requestor,
-                                         size_t max_alloc_words) {
-  if (requestor == this || _promotion_failed) {
+void DefNewGeneration::contribute_scratch(void*& scratch, size_t& num_words) {
+  if (_promotion_failed) {
     return;
   }
-  assert(GenCollectedHeap::heap()->is_old_gen(requestor), "We should not call our own generation");
 
-  /* $$$ Assert this?  "trace" is a "MarkSweep" function so that's not appropriate.
-  if (to_space->top() > to_space->bottom()) {
-    trace("to_space not empty when contribute_scratch called");
-  }
-  */
+  constexpr static size_t MinFreeScratchWords = 100;
```

Comment:
Is there a reason this needs to be a `constexpr static` when a simple `const` seems to be sufficient?

Reply:
Dropped `static`. `constexpr` makes it more a num literal. (The generated release asm is identical anyway.)

Reply:
I think constexpr doesn't do anything here beyond what const would do. I'm not sure about this, but
I think I'd prefer to reserve the use of constexpr only where it (might) matter.

Reply:
Changed to const, but I still believe constexpr should be used for all compile-time known values.